find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
+  max = nil
+  if max_index == 0
+    puts "Array is empty"
+  else
+    max = array[0]
+    index = 0
```

Comment:
max is initialized as nil at the beginning and then reassigned when the else statement gets executed. It might be ok to either initialize it at the beginning as max = array[0] and delete line 9.